### PR TITLE
get time from sqlx-sqlite directly

### LINKED ISSUE #409: [SeaQuery] Support `time` crate on SQLite
Native support of `time` crate has been added in PR https://github.com/launchbadge/sqlx/pull/1865 and released as part of SQLx [0.6.1](https://github.com/launchbadge/sqlx/blob/main/CHANGELOG.md).

Now we could bind `time` on to SQLite query statement and get `time` from SQLite directly.

### PR DESCRIPTION
**Pull request by kyoto7250**
## PR Info

- Closes 
close #409 

## Changes

- [x]  get time from sqlx-sqlite directly

## Breaking Changes
- [x]  In `DateType`,  `time` is no longer compatible with `chrono` .

### CODE REVIEW

File: sea-query-driver/src/sqlx_sqlite.rs [hunk 1]
```diff
@@ -29,10 +29,10 @@ pub fn bind_params_sqlx_sqlite(input: TokenStream) -> TokenStream {
 
     let with_time = if cfg!(feature = "with-time") {
         quote! {
-            v @ Value::TimeDate(_) => query.bind(v.time_as_naive_utc_in_string()),
-            v @ Value::TimeTime(_) => query.bind(v.time_as_naive_utc_in_string()),
-            v @ Value::TimeDateTime(_) => query.bind(v.time_as_naive_utc_in_string()),
-            v @ Value::TimeDateTimeWithTimeZone(_) => query.bind(v.time_as_naive_utc_in_string()),
```
Comment (kyoto7250) [inline sea-query-driver/src/sqlx_sqlite.rs:35]: `time_as_naive_utc_in_string` is still used by `rusqlite` and cannot be removed.

**Review by ikrivosheev — COMMENTED** [review]
@kyoto7250, thank you! Maybe we need to upgrade sqlx version here https://github.com/SeaQL/sea-query/blob/master/sea-query-binder/Cargo.toml#L18 ?

**Comment by ikrivosheev** [comment]
@kyoto7250 some tests failed

**Comment by kyoto7250** [comment]
Ah, This code does not work because this code is not pattern matching anymore.
I didn't notice because the test passes.

` v @ Value::TimeDate(_) => query.bind(v.as_deref())`

**Review by ikrivosheev — APPROVED** [review]
@kyoto7250 thank you! LGTM!

**Comment by kyoto7250** [comment]
@ikrivosheev 

Hmm, we got a runtime error in CI.

Sorry, I will check it.

**Comment by ikrivosheev** [comment]
> @ikrivosheev
> 
> Hmm, we got a runtime error in CI.
> 
> Sorry, I will check it.

I think sqlx convert time object differently than sea-query do it before...

**Comment by kyoto7250** [comment]
It appears that `sqlx v0.6.1` returns `subsecond`, but this behaviour is different from `chrono`.
I think this PR includes `breaking change`.


https://github.com/SeaQL/sea-query/blob/6db27fdc82285a9e585997a762438b5c941b0073/src/value.rs#L436-L437

**Review by billy1624 — COMMENTED** [review]
Hey @kyoto7250, thanks for the contributions!

Yes, this PR contains breaking changes, to be specific, 

- `Value::TimeDate(Option<Box<time::Date>>)` is being serialized and store as String with format
    - old format: `[year]-[month]-[day]`
    - new format: `[year]-[month]-[day]`
    - No Change!
- `Value::TimeTime(Option<Box<time::Time>>)` is being serialized and store as String with format
    - old format: `[hour]:[minute]:[second]`
    - new format: `[hour]:[minute]:[second].[subsecond]`
- `Value::TimeDateTime(Option<Box<PrimitiveDateTime>>)` is being serialized and store as String with format
    - old format: `[year]-[month]-[day] [hour]:[minute]:[second]`
    - new format: `[year]-[month]-[day] [hour]:[minute]:[second].[subsecond]`
- `Value::TimeDateTimeWithTimeZone(Option<Box<OffsetDateTime>>)` is being serialized and store as String with format
    - old format: `[year]-[month]-[day] [hour]:[minute]:[second] [offset_hour sign:mandatory][offset_minute]`
    - new format: `[year]-[month]-[day]T[hour]:[minute]:[second].[subsecond]Z` ([RFC 3339](https://docs.rs/time/0.3.13/time/format_description/well_known/struct.Rfc3339.html))

**Review by ikrivosheev — COMMENTED** [review]
@kyoto7250 hello! Tests are failed: https://github.com/SeaQL/sea-query/runs/7969914195?check_suite_focus=true

File: src/value.rs [hunk 2]
```diff
@@ -432,12 +432,11 @@ pub mod time_format {
...
-        format_description!("[hour]:[minute]:[second]");
+        format_description!("[hour]:[minute]:[second].[subsecond]");
     pub static FORMAT_DATETIME: &[FormatItem<'static>] =
-        format_description!("[year]-[month]-[day] [hour]:[minute]:[second]");
-    pub static FORMAT_DATETIME_TZ: &[FormatItem<'static>] = format_description!(
-        "[year]-[month]-[day] [hour]:[minute]:[second] [offset_hour sign:mandatory][offset_minute]"
-    );
+        format_description!("[year]-[month]-[day] [hour]:[minute]:[second].[subsecond]");
+    pub static FORMAT_DATETIME_TZ: &[FormatItem<'static>] =
+        format_description!("[year]-[month]-[day] [hour]:[minute]:[second].[subsecond]Z");
```
Comment (billy1624) [inline src/value.rs:439]: Hey @kyoto7250, I don't think we need to change this part of the code. This is just for `to_string` on SQL statement and it has nothing to do with value binding. I think we should keep it as-is.

Comment (billy1624) [reply src/value.rs:439]: Thoughts? @kyoto7250

**Comment by kyoto7250** [comment]
Sorry, I am traveling until this week, so I will check it in this weekend.

Thank you for your kind 👍🏽

**Comment by kyoto7250** [comment]
I'm sorry I'm late.
I resolved the conflict.
And I revert 34b96fb,  so could you please review again?

**Comment by ikrivosheev** [comment]
@kyoto7250 hello! Can you resolve merge conflict?

**Comment by ikrivosheev** [comment]
@kyoto7250 any updates?

**Review by ikrivosheev — APPROVED** [review]
@kyoto7250 thank you! LGTM!

**Review by billy1624 — APPROVED** [review]
Thanks!! @kyoto7250

**Review by tyt2y3 — APPROVED** [review]
@ikrivosheev feel free to merge it

**Pull request merged**